nextmv/options.py

```diff
@@ -163,12 +173,15 @@ def __init__(self, *parameters: Parameter):
                 setattr(self, arg, value)
                 continue
 
-            # Finally, attempt to set a default value. This is only allowed
-            # for non-required parameters.
-            if not param.required:
+            # Finally, attempt to set the value of a parameter from the default
+            # value.
+            if param.default is not None:
                 setattr(self, arg, param.default)
                 continue
 
+            if not param.required:
+                continue
+
```

Comment:
I believe the current logic is incorrect. This is what it does today:

* If a parameter is not set and not required, the default is set.

I believe this should be:

* If a parameter is not set and has a default, then the default is set.
* If a parameter is not required and has no default, then the parameter is not set.

Reply:
I just changed this behavior around to be more in line with argparse behavior. To me it seemed pretty simple, and aligns with the first case you are correctly stating. I would personally prefer it, as it is also simpler.

What is the reason to change it? I think I am missing something.

E.g., if I specify some `choices=` on argparse **and** they are _not_ required **and** they do _not_ have a default (i.e., just this: `choices=["a", "b", "c"], type=str`), I get the following when not providing the parameter: `None`. No exception is thrown on accessing the field by name (`args.choice`). 

One thing we had to fix was that providing a `None` _default_ did not work, and I think would again not work with above logic. :thinking: 

I just noticed that the test that I added unfortunately does not cause the exception like normal use would do. I opened a PR to hopefully improve the test and better explain why I have concerns (this would directly affect my use of this functionality - see #43  ). :sweat_smile:

Reply:
From quickly reading this, I think I concur with Marius. We need to be able to set the param to `None`.